### PR TITLE
Build System Cleanup

### PR DESCRIPTION
**Pull request by Diego-Ivan**
This merge request implements:

* The Build.vala.in file has been replaced by a .vapi file and a C header file
* The meson.build file has been split into various files for the src/ subdirectory
* Add the Vala SDK Extension, which provides autocompletion for Flatpak GNOME Builder and Flatpak VS Code
* Add a G_LOG_DOMAIN variable

### CODE REVIEW

**Comment by GeopJr** [comment]
cc: @fortysixandtwo @kop316

File: meson.build [hunk 1]
```diff
@@ -9,25 +9,22 @@ project(
...
+config.set_quoted('PREFIX', get_option('prefix'))
+config.set_quoted('NAME', 'Tooth')
+config.set_quoted('WEBSITE', 'https://github.com/GeopJr/tooth')
+config.set_quoted('SUPPORT_WEBSITE', 'https://github.com/GeopJr/tooth/issues')
+configure_file(output: 'config.h', configuration: config)
+
+add_project_arguments (
+  '--include','config.h',
+  '-DGETTEXT_PACKAGE="@0@"'.format(meson.project_name()),
+  language: 'c'
```
Comment (fortysixandtwo) [inline meson.build:29]: I know `c` was already used before, but I'm wondering whether it shouldn't be `vala` instead 
as it is mentioned as a supported language string in the [docs](https://mesonbuild.com/Reference-manual_functions.html#project)

File: src/Application.vala [hunk 2]
```diff
@@ -219,11 +238,12 @@ namespace Tooth {
...
-				copyright = Build.COPYRIGHT,
-				debug_info = Build.SYSTEM_INFO,
-				developers = developers,
-				artists = artists,
-				translator_credits = Build.TRANSLATOR != " " ? Build.TRANSLATOR : ""
+				copyright = COPYRIGHT,
+				debug_info = print_info (),
+				developers = DEVELOPERS,
+				artists = ARTISTS,
+				// translators: Write you Name<email> here :)
```
Comment (fortysixandtwo) [inline src/Application.vala:245]: typo: s/you/your 
:)

**Review by fortysixandtwo — COMMENTED** [review]
Only really found two nits, lgtm

File: meson.build (hunk 1 above)
Comment (Diego-Ivan) [reply meson.build:29]: Vala is supported, but those arguments are for the C compiler, as it asking to include the C header that was generated using `configure_file`. On the other hand, we will use that C Header in Vala through the VAPI that is included in the project files

**Comment by GeopJr** [comment]
Sorry for the wait!

I brought this up-to-date and organized it slightly more (one meson.build per folder)

**Pull request merged**